README.md

```diff
@@ -1,9 +1,243 @@
+    return [Node("my_node", **kwargs), Node("my_other_node", **kwargs)]
+
+@ros_process.main(prebaked=False)
+def main() -> None:
+    ros_process.load(Node, "_hidden_node")  # or main.load
+    ros_process.spin(graph)  # or main.spin, and it will block!
+
+if __name__ == "__main__":
+    main()
+```
+
+Note the use of ROS 2 node(s) factories, to load (or spin) entire collections at once.
+
+#### Setting up interactive multi-node applications
+
+For interactive applications, you will most likely want to keep automatic spinning in place:
+
+```python
+import logging
+import time
+
+from typing import Any, List
+
+from bdai_ros2_wrappers.futures import wait_for_future
+from bdai_ros2_wrappers.node import Node
+import bdai_ros2_wrappers.process as ros_process
+
+def graph(**kwargs: Any) -> List[Node]:
+    # make sure to forward all keyword arguments for proper loading!
+    return [Node("my_node", **kwargs), Node("my_other_node", **kwargs)]
+
+def work(prompt: str) -> None:
+    def worker() -> None:
+        node = ros_process.node()
+        assert node is not None
+        time.sleep(5.0)  # sleeping is safe!
+        node.get_logger().info(prompt + " done!")
+    executor = ros_process.executor()
+    assert executor is not None
+    future = executor.create_task(worker)  # dispatch worker to executor
+    wait_for_future(future)  # block until worker is done
+
+@ros_process.main()
+def main() -> None:
+    ros_process.load(graph)  # or main.load
+    while True:
+        work(input())
+
+if __name__ == "__main__":
+    main()
+```
+
+#### Setting up command-line single node applications
+
+```python
+import argparse
+import logging
+import time
+
+from typing import Any
+
+from bdai_ros2_wrappers.futures import wait_for_future
+from bdai_ros2_wrappers.node import Node
+import bdai_ros2_wrappers.process as ros_process
+
+class Application(Node):
+
+    def __init__(self, robot_name: str, **kwargs: Any) -> None:
+        super().__init__("app_node", **kwargs)
+        self.robot_name = robot_name
+
+def cli() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser()
+    parser.add_argument("robot_name")
+    parser.add_argument("-v", "--verbose", action="store_true")
+    # define ROS 2 aware process arguments based on CLI arguments
+    parser.set_default(process_args=lambda args: dict(
+        prebaked=False, forward_logging=args.verbose))
+    return parser
+
+@ros_process.main(cli())
+def main(args: argparse.Namespace) -> None:
+    # set process-wide node for log forwarding to take effect
+    with ros_process.managed(Application, args.robot_name) as main.node:
+        if args.verbose:
+            root = logging.getLogger()
+            root.setLevel(logging.INFO)
+        logging.info("Application started!")
+        ros_process.spin()  # or main.spin
+
+if __name__ == "__main__":
+    main()
+```
+
+Note that process arguments are set via CLI. Also, note that a process-wide node is set explicitly (rather than implicitly). 
```

Comment:
In what cases does fine grained control over ROS 2 configuration matter?  Could you expand a little here on why you want to set a process-wide node explicitly?

Reply:
Clarified in https://github.com/bdaiinstitute/ros_utilities/pull/32/commits/efb0db0d1ef57bdecc49aaff0dcbe9222defb1e3.